src/main/java/org/jitsi/jigasi/lobby/Lobby.java

```diff
@@ -0,0 +1,259 @@
+import net.java.sip.communicator.service.protocol.*;
+import net.java.sip.communicator.util.*;
+
+public class Lobby
+        extends AbstractRoom
+        implements ChatRoomInvitationListener
+{
+    /**
+     * The logger.
+     */
+    private final static Logger logger = Logger.getLogger(Lobby.class);
+
+    /**
+     * The name of the property which configured the local region.
+     */
+    public static final String LOCAL_REGION_PNAME
+        = "org.jitsi.jigasi.LOCAL_REGION";
+
+    /**
+     * The name of XMPP feature which states for Jigasi SIP Gateway and can be
+     * used to recognize gateway client.
+     */
+    public static final String SIP_GATEWAY_FEATURE_NAME
+        = "http://jitsi.org/protocol/jigasi";
+
+    /**
+     * The XMPP provider used to join JVB conference.
+     */
+    private ProtocolProviderService xmppProvider;
+
+    /**
+     * The XMPP password used for the call handled by this instance.
+     */
+    private String xmppPassword;
+
+    /**
+     * Display name for lobby room.
+     */
+    private String displayName;
+
+    /**
+     * Room full Jid.
+     */
+    private EntityFullJid roomJid;
+
+    /**
+     * <tt>ProtocolProviderFactory</tt> instance used to manage XMPP accounts.
+     */
+    private ProtocolProviderFactory xmppProviderFactory;
+
+    private CopyOnWriteArrayList<LobbyListener> lobbyListeners;
+    private CallContext callContext;
+
+    public Lobby(ProtocolProviderService protocolProviderService, CallContext context, EntityFullJid jid)
+    {
+        super();
+
+        this.xmppProvider = protocolProviderService;
+
+        this.lobbyListeners = new CopyOnWriteArrayList<>();
+
+        this.roomJid = jid;
+
+        this.callContext = context;
+    }
+
+    public void setDisplayName(String displayName)
+    {
+        this.displayName = displayName;
+    }
+
+    @Override
+    protected void joinRoom(Jid roomJid, byte[] roomPassword) throws OperationFailedException, OperationNotSupportedException
+    {
+        OperationSetMultiUserChat muc
+                = this.xmppProvider.getOperationSet(OperationSetMultiUserChat.class);
+
+        muc.addInvitationListener(this);
+
+        super.joinRoom(roomJid, roomPassword);
+    }
+
+    @Override
+    protected void leaveRoom()
+    {
+        OperationSetMultiUserChat muc
+                = this.xmppProvider.getOperationSet(OperationSetMultiUserChat.class);
+
+        muc.removeInvitationListener(this);
+
+        super.leaveRoom();
+    }
+
+    public void addLobbyListener(LobbyListener lobbyListener)
+    {
+        if (this.lobbyListeners != null)
+        {
+            this.lobbyListeners.add(lobbyListener);
+        }
+    }
+
+    public void removeLobbyListener(LobbyListener lobbyListener)
+    {
+        if (this.lobbyListeners != null)
+        {
+            this.lobbyListeners.remove(lobbyListener);
+        }
+    }
+
+    private void approve(ChatRoomInvitationReceivedEvent chatRoomInvitationReceivedEvent)
+    {
```

Comment:
This is not "approve" ... it just adds a listener for access approved. You can move it directly under invitationReceived.

Reply:
Removed approve and reject and moved the code in the listener methods